toponetx/classes/path_complex.py

```diff
@@ -0,0 +1,657 @@
+        index : bool, default=False
+            If True, return incidence matrix with indices. Else, return incidence matrix without indices.
+
+        Returns
+        -------
+        If `index` is True, return a tuple of (idx_p_minus_1, idx_p, incidence_matrix).
+        If `index` is False, return incidence_matrix.
+        """
+        if rank < 0:
+            raise ValueError(f"input dimension d must be positive integer, got {rank}")
+        if rank > self.dim:
+            raise ValueError(
+                f"input dimenion cannat be larger than the dimension of the complex, got {rank}"
+            )
+        if rank == 0:
+            boundary = sp.sparse.lil_matrix((0, len(self.nodes)))
+            if index:
+                node_index = {
+                    node: i
+                    for i, node in enumerate(sorted(self.nodes, key=lambda x: str(x)))
+                }
+                return {}, node_index, abs(boundary.tocoo())
+            else:
+                return abs(boundary.tocoo())
+        else:
+            idx_p_minus_1, idx_p, values = [], [], []
+            path_minus_1_dict = {
+                path: i for i, path in enumerate(self.skeleton(rank - 1))
+            }  # path2idx dict
+            path_dict = {
+                path: i for i, path in enumerate(self.skeleton(rank))
+            }  # path2idx dict
+            for path, idx_path in path_dict.items():
+                for i, _ in enumerate(path):
+                    boundary_path = path[0:i] + path[(i + 1) :]
+                    if not self._reserve_sequence_order and str(boundary_path[0]) > str(
+                        boundary_path[-1]
+                    ):
+                        boundary_path = boundary_path[::-1]
+                    boundary_path = tuple(boundary_path)
+                    if boundary_path in self._allowed_paths:
+                        idx_p_minus_1.append(path_minus_1_dict[boundary_path])
+                        idx_p.append(idx_path)
+                        values.append((-1) ** i)
+            boundary = sp.sparse.coo_matrix(
+                (values, (idx_p_minus_1, idx_p)),
+                dtype=np.float32,
+                shape=(
+                    len(path_minus_1_dict),
+                    len(path_dict),
+                ),
+            )
+        if index:
+            if signed:
+                return (
+                    path_minus_1_dict,
+                    path_dict,
+                    boundary,
+                )
+            else:
+                return (
+                    path_minus_1_dict,
+                    path_dict,
+                    abs(boundary),
+                )
+        else:
+            if signed:
+                return boundary
+            else:
+                return abs(boundary)
+
+    def coincidence_matrix(self, rank: int, signed: bool = True, index: bool = False):
```

Comment:
Since this matrix is just the transponse of the earlier one then I do not think it needed as a separate method. WE did not include these methods in other complexes.

Reply:
I just removed the `coincidence_matrix()`. I notice that `SimplicialComplex()` still has this method.